### PR TITLE
Hide warning when booking already exists

### LINKED ISSUE #3987: Hide booking warning in case there is a booking for one of the managers during that period.

### PR DESCRIPTION
**Pull request by panagiotappl**
closes #3987

### CODE REVIEW

File: indico/modules/events/management/controllers/settings.py [hunk 1]
```diff
@@ -41,7 +46,21 @@ def _check_access(self):
         RHManageEventBase._check_access(self)  # mainly to trigger the legacy "event locked" check
 
     def _process(self):
-        return WPEventSettings.render_template('settings.html', self.event, 'settings')
+        # Check if any of the managers of the event already have a booking that overlaps with the event datetime
+        manager_ids = [p.user.id for p in self.event.acl_entries if p.user]
```
Comment (ThiefMaster) [inline indico/modules/events/management/controllers/settings.py]: that's everyone in the ACL, not just managers. the correct way to get only the managers is:

```python
manager_ids = {p.user.id
               for p in self.event.acl_entries
               if p.type == PrincipalType.user and p.has_management_permission()}
```

File: indico/modules/events/management/controllers/settings.py [hunk 2]
```diff
@@ -41,7 +46,21 @@ def _check_access(self):
...
+        has_overlap = (ReservationOccurrence.query
+                       .filter(ReservationOccurrence.is_valid,
+                               db.or_(Reservation.booked_for_id.in_(manager_ids),
+                                      Reservation.created_by_id.in_(manager_ids)),
+                               db_dates_overlap(ReservationOccurrence,
+                                                'start_dt', self.event.start_dt_local,
+                                                'end_dt', self.event.end_dt_local),
+                               Reservation.room_id == self.event.room.id,
```
Comment (ThiefMaster) [inline indico/modules/events/management/controllers/settings.py]: I think this breaks the settings page if the event has no room assigned...

File: indico/modules/events/management/templates/settings.html [hunk 3]
```diff
@@ -5,7 +5,8 @@
 
 {% block content %}
     {{ template_hook('event-actions', event=event) }}
-    {%- if indico_config.ENABLE_ROOMBOOKING and not event.has_ended and event.room and not event.room_reservation_links -%}
+    {%- if indico_config.ENABLE_ROOMBOOKING and not event.has_ended and event.room and not event.room_reservation_links
+    and not has_overlap -%}
```
Comment (ThiefMaster) [inline indico/modules/events/management/templates/settings.html]: that's a weird indentation level for wrapping this. how about moving this logic into the RH class and just check e.g. `if show_no_booking_warning` here?

File: indico/modules/events/management/controllers/settings.py [hunk 4]
```diff
@@ -11,13 +11,18 @@
...
+from indico.core.db import db
+from indico.core.db.sqlalchemy.util.queries import db_dates_overlap
 from indico.modules.events.management.controllers.base import RHManageEventBase
 from indico.modules.events.management.forms import (EventClassificationForm, EventContactInfoForm, EventDataForm,
                                                     EventDatesForm, EventLocationForm, EventPersonsForm)
 from indico.modules.events.management.util import flash_if_unregistered
 from indico.modules.events.management.views import WPEventSettings, render_event_management_header_right
 from indico.modules.events.operations import update_event
 from indico.modules.events.util import track_time_changes
+from indico.modules.rb import Room
```
Comment (mic4ael) [inline indico/modules/events/management/controllers/settings.py]: It works just by coincidence. Use `from indico.modules.rb.models.rooms import Room`

**Pull request merged**